test/karma.config.mjs

```diff
@@ -0,0 +1,167 @@
+  const browsers = (CI ?
+    ['ChromeCI', 'FirefoxCI'] :
+    [...(!VISUAL_TEST_CONFIG.debug && !VISUAL_TEST_CONFIG.recreate ? ['ChromeCI', 'FirefoxCI'] : []), 'Chrome', 'Firefox'])
+    .filter(browser => BROWSERS.some(b => browser.toLowerCase().startsWith(b)));
+  if (VISUAL_TEST_CONFIG.debug || VISUAL_TEST_CONFIG.recreate) {
+    if (browsers.length > 1) {
+      console.warn(chalk.yellow(`Debugging/recreating visual tests is allowed ONLY when running tests in a single browser`));
+    }
+    if (CI) {
+      throw new Error(chalk.red(`Debugging/recreating visual refs is banned in CI`));
+    }
+  }
+  if (browsers.length === 0) {
+    throw new Error(`no browsers to launch, input: ${BROWSERS}`);
+  }
+  
+  config.set({
+    plugins: [
+      'karma-jasmine',
+      'karma-qunit',
+      'karma-chrome-launcher',
+      'karma-firefox-launcher',
+      'karma-coverage',
+      'karma-spec-reporter'
+    ],
+
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    basePath: '../',
+    autoWatch: false,
+    singleRun: true,
+    pingTimeout: CI ? 10000 : 2000,
+    browserDisconnectTimeout: CI ? 10000 : 2000,
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: [/*'jasmine',*/ 'qunit'],
+    browsers,
+    customLaunchers: {
+      ChromeCI: {
+        base: 'Chrome',
+        flags: [
+          '--no-sandbox',
+          '--disable-gpu',
+          '--disable-translate',
+          '--disable-extensions',
+          '--remote-debugging-port=9222',
+        ],
+      },
+      FirefoxCI: {
+        base: 'Firefox'
+      }
+    },
+
+    reporters: ['spec', 'coverage'],
+    coverageReporter: {
+      reporters: [
+        { type: 'lcov', dir: '.nyc_output/' },
+        // Karma uses subdirs by default to account for multiple browsers.
+        // For the JSON file, it's important we disable 'subdir' so that
+        // the 'nyc report' command can pick this up when combining code
+        // coverage with the Node.js test run.
+        { type: 'json', dir: '.nyc_output/', subdir: '.' }
+      ]
+    },
+
+    // list of files / patterns to load in the browser
+    files: [
+      { pattern: 'test/fixtures/**/*', included: false, served: true, watched: false, nocache: false },
+      { pattern: 'test/lib/*.js', included: true, served: true, watched: true, nocache: false },
+      { pattern: 'test/lib/tests.css', included: true, served: true, watched: true, nocache: false }, // qunit only
+      { pattern: 'test/visual/golden/**/*', included: false, served: true, watched: false, nocache: false },
+      { pattern: 'test/visual/assets/**/*', included: false, served: true, watched: false, nocache: false },
+
+      { pattern: 'dist/index.js', type: 'js', included: true, served: true, watched: true, nocache: true },
+      { pattern: 'dist/index.js.map', included: false, served: true, watched: false, nocache: true },
```

Comment:
follow up will also bundle src code directly so we could import modules directly even if not exposed/internal